flink-runtime/src/main/java/org/apache/flink/runtime/security/token/hadoop/HadoopDelegationTokenReceiver.java

```diff
@@ -19,46 +19,43 @@
 package org.apache.flink.runtime.security.token.hadoop;
 
 import org.apache.flink.annotation.Internal;
-import org.apache.flink.runtime.security.token.DelegationTokenContainer;
-import org.apache.flink.util.InstantiationUtil;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.runtime.security.token.DelegationTokenReceiver;
 
 import org.apache.hadoop.security.Credentials;
 import org.apache.hadoop.security.UserGroupInformation;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-/** Delegation token updater functionality. */
+/** Hadoop delegation token receiver base class. */
 @Internal
```

Comment:
Could I use this class for the HMSDelegationTokenReceiver implementation?

Reply:
You could but since it's `@Internal` it's subject to change.